src/ansi.rs

```diff
@@ -189,7 +189,7 @@ fn find_ansi_code_exclusive(it: &mut Peekable<CharIndices>) -> Option<(usize, us
 
 /// Helper function to strip ansi codes.
 #[cfg(feature = "alloc")]
-pub fn strip_ansi_codes(s: &str) -> Cow<str> {
+pub fn strip_ansi_codes(s: &str) -> Cow<'_, str> {
```

Comment:
Since this is independent of your other changes, please isolate it in a separate commit.